components/user_settings/sidebar/user_settings_sidebar.tsx

```diff
@@ -4,829 +4,72 @@
-                                defaultMessage='Alphabetically'
-                            />
-                        </label>
-                    </div>
-                    <div className='mt-5'>
-                        <FormattedMessage
-                            id='user.settings.sidebar.sortDesc'
-                            defaultMessage='Sort channels alphabetically, or by most recent post.'
-                        />
-                    </div>
-                </fieldset>,
-            );
-
-            inputs.push(<hr key='divider'/>);
-
-            inputs.push(
-                <fieldset key='unreadOption'>
-                    <div className='checkbox'>
-                        <label>
-                            <input
-                                id='unreadAtTopOption'
-                                type='checkbox'
-                                checked={this.state.settings.unreadsAtTop === 'true'}
-                                onChange={(e) => this.updateSetting(Settings.UnreadsAtTop, (e.target.checked).toString())}
-                            />
-                            <FormattedMessage
-                                id='user.settings.sidebar.unreads'
-                                defaultMessage='Unreads grouped separately'
-                            />
-                        </label>
-                    </div>
-                    <div className='mt-5'>
-                        <FormattedMessage
-                            id='user.settings.sidebar.unreadsDesc'
-                            defaultMessage='Group unread channels separately until read.'
-                        />
-                    </div>
-                </fieldset>,
-            );
-
-            inputs.push(<hr key='groupingDivider'/>);
-
-            inputs.push(
-                <fieldset key='favoriteOption'>
-                    <div className='checkbox'>
-                        <label>
-                            <input
-                                id='favoriteAtTopOption'
-                                type='checkbox'
-                                checked={this.state.settings.favoriteAtTop === 'true'}
-                                onChange={(e) => this.updateSetting(Settings.FavoriteAtTop, (e.target.checked).toString())}
-                            />
-                            <FormattedMessage
-                                id='user.settings.sidebar.favorites'
-                                defaultMessage='Favorites grouped separately'
-                            />
-                        </label>
-                    </div>
-                    <div>
-                        <br/>
-                        <FormattedMessage
-                            id='user.settings.sidebar.favoritesDesc'
-                            defaultMessage='Channels marked as favorites will be grouped separately.'
+export default function UserSettingsSidebar(props: Props): JSX.Element {
```

Comment:
Turns out this thing is incredibly simple now that there's only two settings and they both manage their own saving and rendering.